src/components/vega-wallet/vega-wallet-container.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import {
+  AppStateActionType,
+  useAppState,
+  VegaWalletStatus,
+} from "../../contexts/app-state/app-state-context";
+import {
+  Errors,
+  vegaWalletService,
+} from "../../lib/vega-wallet/vega-wallet-service";
+import { VegaWalletForm } from "./vega-wallet-form";
+import { VegaWalletNotRunning } from "./vega-wallet-not-running";
+
+export const VegaWalletFormContainer = () => {
+  const { appState, appDispatch } = useAppState();
+  const [isVegaWalletRunning, setIsVegaWalletRunning] = React.useState(
+    appState.vegaWalletStatus !== VegaWalletStatus.None
+  );
+
+  React.useEffect(() => {
+    const run = async () => {
+      const [version] = await vegaWalletService.getVersion();
+      setIsVegaWalletRunning(version !== Errors.SERVICE_UNAVAILABLE);
+    };
+    run(); // run it first so that the intial state is correct
+    const interval = setInterval(run, 2000); // then pull every 2 seconds in case the wallet is started/stopped while the modal is up
+
+    return () => clearInterval(interval);
```

Comment:
Should probably also clear interval when the wallet is detected, not just on unmount

Reply:
The reason I'm clearing it is every time the wallet service stops/starts while the popup is open, I want to show the appropriate state of the modal

Reply:
Okay that makes sense 👍